doc/src/manual/interfaces.md

```diff
@@ -739,3 +739,67 @@ yields another `SparseVecStyle`, that its combination with a 2-dimensional array
 yields a `SparseMatStyle`, and anything of higher dimensionality falls back to the dense arbitrary-dimensional framework.
 These rules allow broadcasting to keep the sparse representation for operations that result
 in one or two dimensional outputs, but produce an `Array` for any other dimensionality.
+
+## [IO Stream](@id man-interface-iostream)
+
+| Required methods                          | Brief description                                                                               |
+|:----------------------------------------- |:----------------------------------------------------------------------------------------------- |
+| `unsafe_read(io, ::Ptr{UInt8}, ::UInt)`   | Copy bytes from the `IO` to a pointer. |
+| `unsafe_write(io, ::Ptr{UInt8}, ::UInt)`  | Copy bytes from a pointer to the `IO`. |
+| `eof(io)`                                 | Whether the IO stream is at the end.   |
```

Comment:
```suggestion
| Required methods                          | Brief description                             |
|:----------------------------------------- |:----------------------------------------------|
| `read(io, ::Type{UInt8})`                 | Read a byte from the stream.                  |
| `write(io, ::UInt8)`                      | Write a byte to the stream.                   |
| `eof(io)`                                 | Whether the IO stream is at the end.          |
```

Reply:
wait... I'm very confused, I tested with only `read` and `write` and it didn't work, that's why I have the `unsafe_` methods (obviously being able to use `read` and `write` is vastly preferable.  Am I missing something here?  If you are convinced this should work, I can probably figure out a MWE and show you.

Reply:
Probably doesn't get much testing currently, but your `IORecorder` type seems like a candidate to also run as a test (or doctest)

Reply:
I tested with `IORecorder(stdin)`, and it worked